### DISCUSSION TITLE
impossible to maximize each part of dockge

### DISCUSSION THREAD

**Discussion by miltuss** [louislam/dockge#493] [Feature Request]
Dockge is divided into three horizontal windows: The list of stacks, container management and the docker-compose editor. 

 This organization drastically reduces the width of each window, which is not ergonomic at all.  In addition, it is impossible to enlarge one of these windows to have a clearer display of one of the parts.  With Dockge we are currently forced to work with a screen divided into three parts.

 It would be so much better if it was possible to enlarge each part to work on it better.

**Comment by Isotop7** [comment]
This would be really cool! I use the `compose` log output in `dockge` quiet often but if I am using a fairly small screen, the `compose.yml`-Preview should minimize in order to make more room for the output.

+1

**Comment by docker-master** [comment]
I like this too

**Comment by dominikborkowski** [comment]
First time trying out dockge, with hopes of replacing vscode server as my main means of editing docker-compose.yml files. Even on high res monitor having the 1/3rd of page for actual editor means that a large number of single lines from docker-compose files span multiple lines: virtually every of the traefik labels spans four lines.

I'd love to see the layout re-arranged, or ability to dynamically re-arrange/resize components. For example, left most column with list of docker containers could be scaled down to the length of the longest name: as it stands, it's wider, and feels like it's a lot of wasted space that could be used towards the editor.

Cheers!

**Comment by alikome** [comment]
I just installed and setup Dockge, great tool tbf, but i came here to post this exact suggestion.

At least for the yaml text editor, a maximize button would be a game changer.

Another solution would be to put the yaml editor in the bottom and give it a full width.

**Reply by miltuss** [reply]
Unfortunately, @louislam isn't listening to his community. I'd be surprised if he finally takes action and fixes Dockge's problems.

**Reply by Isotop7** [reply]
@miltuss I would also appreciate this feature but this is a free product developed as a hobby. You are free to submit a pull request with the required changes but having this kind of attitude will lead more open source developers to abandon their work.

**Reply by miltuss** [reply]
I completely understand that. But @louislam doesn't respect his community and never responds to messages, and doesn't accept PRs that have been hard-worked by contributors. I don't compare him to other OpenSource developers.
(I'm not the only one saying this)

**Reply by Celer21** [reply]
@miltuss  You can fork this project, merge PRs and be better than @louislam. Or you another solution...
This is personal project of @louislam so its his desision how many time is he spending on this project.

**Reply by alikome** [reply]
Tbf it's a free app, i actually thought about adding it myself but then i read the contribution guide and got discouraged immediately, because there's a very high chance the pr won't be accepted

**Reply by miltuss** [reply]
Yes, the main developer makes things complicated and never responds to messages. Contributions are not accepted and nothing changes!

**Comment by molivier-io** [comment]
This is not a definitive solution, but someone tweaked the CSS code with ublock origin in order to force wider panels. Be careful about changing `DOCKGE_URL` with your real instance address.

These settings can be forced one or another plugin able to hack websites CSS.

https://github.com/louislam/dockge/discussions/31#discussioncomment-10820324

**Reply by docker-master** [reply]
My requirement is the comment of the env file, which is not wide enough. With AI, I just ask him to rewrite it with 58 characters. The requirement is solved.

**Reply by molivier-io** [reply]
Ok. For your information, CSS changes give this 2 columns layout in any view:

<img width="1200" height="2621" alt="dockge-resized" src="https://github.com/user-attachments/assets/dc15e168-a4c3-4dad-a4c0-95fbbc7b29ed" />
